### PR TITLE
Feature: Set receive amount & label

### LINKED ISSUE #144: Ability to set amount to receive
Add number field to `ReceiveDetails` screen to set amount to receive which would update URI as specified in BIP21 and thus generated QR code.

### PR DESCRIPTION
**Pull request by pajasevi**
This PR adds ability to set amount and label on receive screen (closes #144). Produced QR code refreshes automatically on change. Right now only english translations were added for these new fields.
simulator screen shot - iphone x - 2018-12-21 at 13 31 20
simulator screen shot - iphone x - 2018-12-21 at 13 33 04

Let me know if there's anything I can fix or adjust in this.

### CODE REVIEW

**Comment by ncoelho** [comment]
This is a great. 

I would try to make this screen less cluttered. For instance...

Main view:
- QR code
- Address
- Share
**- Create invoice (new option)**

Create invoice screen:
- Amount
- Memo
- Create button

And after goes to a new invoice screen.

@pajasevi Do you mind if I mockup something for you on this?

also @marcosrdz does this breaks or changes the invoice screen you are working on?

**Comment by pajasevi** [comment]
Good idea, although I would be careful with the word "invoice" as it could be confused with LN invoice maybe? I don't have a strong opinion on this.

A mockup would be great.

**Comment by Overtorment** [comment]
bw - receive btc - peek 2018-12-21 22-27

Good job overall! But needs some improvements:

1) Layout breaks on my Android emulator when keyboard is shown
2) Feels like QR is re-rendered on each keypress, and it feels quite laggy, it just freezes for seconds. Better use other callback (if RN has any for text inpputs?) or implement some kind of intelligent delay-wait before actually redrawing the QR.

With that fixed I'd prefer to merge ASAP. We can always improve later.

**Comment by pajasevi** [comment]
@Overtorment Thanks for the feedback. Both issues could be fixed if we would implement separate view for setting of amount and label like @ncoelho proposed.

Second issue can be fixed by using onBlur event, will fix that soon.

**Comment by ncoelho** [comment]
So I made this one, I think it is self explanatory but here it goes the thinking behind it:

- View remains simple for basic address copy or QR sharing
- If the users wants to add an amount or memo, there's an option for that near the share call to action
- We have now a new view, where we reuse the price input and text input from the styleguide
- User "creates" (or other expression) the new receive share view with the new added info
- User can share or exit and start from scratch

screenshot 2018-12-22 at 17 28 58

Here's how the flow should be:
https://overflow.io/s/9DM1HL/

@pajasevi I sent you a message to give you access to zeplin where all the elements are with assets export, etc

**Comment by pajasevi** [comment]
Thanks @ncoelho for the mock, I created the screen:

ezgif-2-dc353b22e2a7

Please review if this is correct or if it needs some more adjustments.

**Comment by pajasevi** [comment]
@ncoelho Thanks for the feedback. I fixed the share button and BTC input.

**Comment by Overtorment** [comment]
bug with scanqr

File: screen/receive/receiveAmount.js [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+  }
+
+  renderWithSetAmount() {
+    return (
+      <View style={{ justifyContent: 'space-between' }}>
+        <BlueText style={{ color: '#0c2550', fontWeight: '600', textAlign: 'center', paddingBottom: 24 }} numberOfLines={1}>
+          {this.state.label}
+        </BlueText>
+        <View style={{ justifyContent: 'center', alignItems: 'center', paddingHorizontal: 16 }}>
+          <QRCode
```
[the remark this answers is not in the document]
Comment (pajasevi) [reply screen/receive/receiveAmount.js]: Updated to use QRFast

File: screen/receive/receiveAmount.js [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+            content={bip21.encode(this.state.address, { amount: this.state.amount, label: this.state.label })}
+            size={(is.ipad() && 300) || 300}
```
[the remark this answers is not in the document]
Comment (pajasevi) [reply screen/receive/receiveAmount.js]: Fixed

**Comment by Overtorment** [comment]
https://github.com/BlueWallet/BlueWallet/pull/182

**Pull request closed without merging**